ledger/accountdb_test.go

```diff
@@ -356,19 +356,134 @@ func TestAccountDBRound(t *testing.T) {
 	require.NoError(t, err)
 	checkAccounts(t, tx, 0, accts)
 
+	ctbsList, randomCtbs := randomCreatables()
+	expectedDbImage := make(map[basics.CreatableIndex]modifiedCreatable)
+	
 	for i := 1; i < 10; i++ {
+		ctbsWithDeletes := randomCreatableSampling(i, ctbsList, randomCtbs, expectedDbImage)
 		updates, newaccts, _ := randomDeltas(20, accts, 0)
 		accts = newaccts
-		err = accountsNewRound(tx, updates, nil)
+		err = accountsNewRound(tx, updates, ctbsWithDeletes)
 		require.NoError(t, err)
 		err = totalsNewRounds(tx, []map[basics.Address]accountDelta{updates}, []AccountTotals{AccountTotals{}}, []config.ConsensusParams{proto})
 		require.NoError(t, err)
 		err = updateAccountsRound(tx, basics.Round(i), 0)
 		require.NoError(t, err)
 		checkAccounts(t, tx, basics.Round(i), accts)
+		checkCreatables(t, tx, i, expectedDbImage)
 	}
 }
 
+// checkCreatables compares the expected database image to the actual databse content
+func checkCreatables(t *testing.T,
+	tx *sql.Tx, iteration int,
+	expectedDbImage map[basics.CreatableIndex]modifiedCreatable) {
+
+	stmt, err := tx.Prepare("SELECT asset, creator, ctype FROM assetcreators")
+	require.NoError(t, err)
+
+	defer stmt.Close()
+	rows, err := stmt.Query()
+	if err != sql.ErrNoRows {
+		require.NoError(t, err)
+	}
+	defer rows.Close()	
+	counter := 0
+	for rows.Next() {
+		counter++
+		mc := modifiedCreatable{}
+		var buf[]byte
+		var asset basics.CreatableIndex
+		err := rows.Scan(&asset, &buf, &mc.ctype)
+		require.NoError(t, err)
+		copy(mc.creator[:], buf)
+
+		require.NotNil(t, expectedDbImage[asset])
+		require.Equal(t, mc.creator, expectedDbImage[asset].creator)
+		require.Equal(t, mc.ctype, expectedDbImage[asset].ctype)
+		require.Equal(t, expectedDbImage[asset].created, true)
+	}
+	require.Equal(t, len(expectedDbImage), counter)
+}
+
+// randomCreatableSampling sets elements to delete from previous iteration
+// It consideres 10 elements in an iteration.
+// loop 0: returns the first 10 elements
+// loop 1: returns: * the second 10 elements
+//                  * random sample of elements from the first 10: created changed from true -> false
+// loop 2: returns: * the elements 20->30
+//                  * random sample of elements from 10->20: created changed from true -> false
+func randomCreatableSampling(iteration int, crtbsList[]basics.CreatableIndex,
+	creatables map[basics.CreatableIndex]modifiedCreatable,
+	expectedDbImage map[basics.CreatableIndex]modifiedCreatable) (
+	map[basics.CreatableIndex]modifiedCreatable) {
+
+	numElementsPerSegement := 10
```

Comment:
could you move the `numElementsPerSegement` to be an input parameter to the function ?

Reply:
done.